packages/artifact/src/internal/download-http-client.ts

```diff
@@ -0,0 +1,189 @@
+import * as fs from 'fs'
+import * as zlib from 'zlib'
+import {
+  getArtifactUrl,
+  getRequestOptions,
+  isSuccessStatusCode,
+  isRetryableStatusCode,
+  createHttpClient
+} from './utils'
+import {URL} from 'url'
+import {ListArtifactsResponse, QueryArtifactResponse} from './contracts'
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
+import {HttpManager} from './http-manager'
+import {DownloadItem} from './download-specification'
+import {
+  getDownloadFileConcurrency,
+  getRetryWaitTimeInMilliseconds
+} from './config-variables'
+import {warning} from '@actions/core'
+
+export class DownloadHttpClient {
+  // http manager is used for concurrent connection when downloading mulitple files at once
+  private downloadHttpManager: HttpManager
+
+  constructor() {
+    this.downloadHttpManager = new HttpManager()
+  }
+
+  /**
+   * Gets a list of all artifacts that are in a specific container
+   */
+  async listArtifacts(): Promise<ListArtifactsResponse> {
+    const artifactUrl = getArtifactUrl()
+    this.downloadHttpManager.createClients(1)
```

Comment:
I still don't think this is necessary - if we know how many clients we need (from the `DOWNLOAD_CONCURRENCY`), then we should just create those clients in the constructor.  Just initiate the httpmanager in constructor with `createClients(DOWNLOAD_CONCURRENCY)`.

Then we will not recreate http clients in each method - for `listArtifacts()` just take one from the manager and user it.

Reply:
I overhauled the `HttpManager` after our discussions yesterday. All instantiation of clients now happens inside the manager and the number of clients is equal to the upload or download concurrency.

If a call needs to be made without `keep-alive`, the first client in the manager is used